src/stores/datStore.ts

```diff
@@ -77,44 +110,105 @@ export default class DatStore {
     getRecentTrades()
      */
 
-    async estimateBuyValue(
-        datAddress: string,
-        currencyValue: BigNumber
-    ): Promise<BigNumber> {
+    @action setDatParams(datAddress: string, updated) {
+        if (!this.datParams[datAddress]) {
+            this.datParams[datAddress] = {};
+        }
+
+        this.datParams[datAddress] = {
+            ...this.datParams[datAddress],
+            ...updated,
+        };
+    }
+
+    @action updatePrice(datAddress: string, newPrice: BigNumberCached) {
         const { providerStore } = this.rootStore;
-        const dat = providerStore.getContract(
+        const hasExistingValue =
+            this.datParams[datAddress] &&
+            this.datParams[datAddress].currentPrice;
+
+        // Don't update if stale
+        if (
+            hasExistingValue &&
+            providerStore.isFresher(
+                newPrice.blockNumber,
+                this.datParams[datAddress].currentPrice.blockNumber
+            )
+        ) {
+            this.setDatParams(datAddress, {
+                currentPrice: newPrice,
+            });
+        } else if (!hasExistingValue) {
+            this.setDatParams(datAddress, {
+                currentPrice: newPrice,
```

Comment:
Maybe I'm reading this wrong but seems like both branches of this conditional are doing the same thing?

Reply:
That's not the case, but I agree it's confusing.

We update the existing value to the new one ONLY if the new one is fresher.

If there is no existing value, go ahead and store it.

Reply:
Ah, I see.  I think I am conditioned (see what I did there) to thinking that each branch is a different action.  I guess it comes down to style but I think it would be clearer to me if, since the action is the same in both cases, that the check was added as an "OR" . . . in other words, no else, just one IF that checks for the possible ways that the condition is met